core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -10045,27 +10011,137 @@ protected void addInitialAdminData(boolean addAdmin, boolean initialSetup) throw
+                // Assign the admin group to the admin role.
+                this.updateGroupListOfHybridRole(adminRoleName, null, new String[]{adminRoleName});
+            } catch (Exception e) {
+                String message = "Admin role has not been created. " +
+                        "Error occurs while creating Admin role in primary user store.";
+                String warnMessage = String.format("Hybrid Admin Role :%s is already added. Hence, continue " +
+                        "without adding the hybrid role.", adminRoleName);
+                handleInitialSetupException(initialSetup, e,
+                        ERROR_CODE_DUPLICATE_WHILE_ADDING_A_HYBRID_ROLE.getCode(), message, warnMessage);
+            }
+        } else {
+            String message = "Admin group can not be created in primary user store. " +
+                    "Add-Admin has been set to false. " +
+                    "Please pick a Group name which is exist in the primary user store as Admin group";
+            handleInitialSetup(initialSetup, message);
+        }
+        return created;
+    }
+
+    /**
+     * Handles error scenarios in the initial set up when admin accounts are created.
+     *
+     * @param initialSetup         A flag to indicate whether this is the initial set up of the server.
+     * @param message              The error message to be used in the exception or error log.
+     * @throws UserStoreException  If this method is called during the initial set up.
+     */
+    private void handleInitialSetup(boolean initialSetup, String message)
+            throws UserStoreException {
+
+        // The existing code is being moved to a separate method for better readability
+        // and to avoid code duplication.
+        if (initialSetup) {
+            throw new UserStoreException(message);
+        } else {
+            log.error(message);
+        }
+    }
+
+    /**
+     * Handles caught exception scenarios in the initial set up when admin accounts are created.
+     *
+     * @param initialSetup         A flag to indicate whether this is the initial set up of the server.
+     * @param e                    The exception being handled.
+     * @param errorCode            The error code to be checked for in the exception being handled.
+     * @param message              The message to be printed with the exception or error log.
+     * @param warnMessage          The warn message to be printed if the error code of the exception.
+     *                             matches the error code being checked for.
+     * @throws UserStoreException  If this is called during the initial set up.
+     */
+    private void handleInitialSetupException(boolean initialSetup, Exception e, String errorCode, String message,
+                                             String warnMessage)
+            throws UserStoreException {
+
+        // The existing code is being moved to a separate method for better readability
```

Comment:
why do we have this comment

Reply:
Removed with [22b7042](https://github.com/wso2/carbon-kernel/pull/3996/commits/22b70425cbff6e92fb728408e8d629743a1fb001).